src/app/stack/stack-details/stack-details.component.ts

```diff
@@ -227,7 +234,100 @@ export class StackDetailsComponent implements OnChanges {
         this.componentFilterBy = filterBy.filterBy;
     }
 
-    constructor(private stackAnalysisService: StackAnalysesService) { }
+    modalRef: BsModalRef;
+    token: string;
+    tokenerror: TokenErrorModel = {
+        status: false,
+        type: '',
+        length: 0
+    };
+    tokenErrorStatus: boolean = true;
+    tokenDetail: TokenDetailModel = {
+        id: '',
+        status: 'freetier'
+    }
+    tokenAlertsMessage: string;
+
+
+    observableToken: Observable<any>;
+
+    constructor(private stackAnalysisService: StackAnalysesService, private modalService: BsModalService) { }
+
+    async submitToken() {
+        if (!this.tokenErrorStatus) {
+            await this.stackAnalysisService.putToken(this.getBaseUrl(this.stack), '4c2c13be-2f0b-44a0-9e86-18abb3c60a6b', this.token, this.gatewayConfig)
```

Comment:
Why `4c2c13be-2f0b-44a0-9e86-18abb3c60a6b` is hardcoded?

Reply:
removed